File: src/main/java/com/aiproxyoauth/server/HealthHandler.java
package com.aiproxyoauth.server;

import com.aiproxyoauth.util.ProxyVersion;
import io.javalin.http.Context;
import io.javalin.http.Handler;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.function.LongSupplier;

public class HealthHandler implements Handler {
    private static final String SERVICE_NAME = "AIProxyOauth";

    private final LongSupplier nanoTime;
    private final long startedAtNanos;

    public HealthHandler() {
        this(System::nanoTime, System.nanoTime());
    }

    HealthHandler(LongSupplier nanoTime, long startedAtNanos) {
        this.nanoTime = nanoTime;
        this.startedAtNanos = startedAtNanos;
    }

    @Override
    public void handle(Context ctx) {
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("ok", true);
        response.put("service", SERVICE_NAME);
        response.put("version", ProxyVersion.get());
        response.put("uptime_seconds", uptimeSeconds());

        JsonHelper.toJsonResponse(ctx, response);
    }

    private long uptimeSeconds() {
        long elapsedNanos = Math.max(0L, nanoTime.getAsLong() - startedAtNanos);
        return elapsedNanos / 1_000_000_000L;
    }
}

File: src/main/kotlin/de/moritzf/proxy/server/HealthHandler.kt
package de.moritzf.proxy.server

import de.moritzf.proxy.util.ProxyVersion
import io.javalin.http.Context
import io.javalin.http.Handler
import java.util.function.LongSupplier

class HealthHandler(
    private val nanoTime: LongSupplier = LongSupplier { System.nanoTime() },
    private val startedAtNanos: Long = System.nanoTime(),
) : Handler {
    override fun handle(ctx: Context) {
        val response = linkedMapOf<String, Any>(
            "ok" to true,
            "service" to SERVICE_NAME,
            "version" to ProxyVersion.get(),
            "uptime_seconds" to uptimeSeconds(),
        )

        JsonHelper.toJsonResponse(ctx, response)
    }

    private fun uptimeSeconds(): Long {
        val elapsedNanos = (nanoTime.asLong - startedAtNanos).coerceAtLeast(0L)
        return elapsedNanos / 1_000_000_000L
    }

    private companion object {
        private const val SERVICE_NAME = "AIProxyOauth"
    }
}
